File: perf/benchmarks/micro/math_loop/math_loop.go
// Math-intensive loop — std math micro-benchmark (Go).
package main

import (
	"fmt"
	"math"
	"os"
	"strconv"
	"time"
)

func main() {
	n := int64(5_000_000)
	if len(os.Args) > 1 {
		v, _ := strconv.ParseInt(os.Args[1], 10, 64)
		n = v
	}
	t0 := time.Now()
	s := 0.0
	inv := 1.0 / float64(n)
	for i := int64(0); i < n; i++ {
		x := float64(i) * inv
		s += math.Sin(x)*math.Cos(x) + math.Sqrt(x+1.0) - math.Exp(-x)
	}
	ms := float64(time.Since(t0).Nanoseconds()) / 1_000_000.0
	cs := int64(math.Round(s * 1e6))
	fmt.Printf("CHECKSUM:%d\n", cs)
	fmt.Printf("ELAPSED_MS:%.6f\n", ms)
}

File: perf/benchmarks/micro/math_loop/math_loop.rs
// Math-intensive loop — std math micro-benchmark (Rust).
use std::env;
use std::time::Instant;

fn main() {
    let args: Vec<String> = env::args().collect();
    let n: i64 = if args.len() > 1 { args[1].parse().unwrap_or(5_000_000) } else { 5_000_000 };
    let t0 = Instant::now();
    let mut s: f64 = 0.0;
    let inv = 1.0 / n as f64;
    for i in 0..n {
        let x = i as f64 * inv;
        s += x.sin() * x.cos() + (x + 1.0).sqrt() - (-x).exp();
    }
    let ms = t0.elapsed().as_nanos() as f64 / 1_000_000.0;
    let cs = (s * 1e6).round() as i64;
    println!("CHECKSUM:{}", cs);
    println!("ELAPSED_MS:{:.6}", ms);
}
